src/platform_impl/windows/event_loop.rs

```diff
@@ -1533,6 +1541,38 @@ unsafe extern "system" fn public_window_callback<T: 'static>(
                                 }
                             }
                         }
+                        winuser::PT_MOUSE |
+                        winuser::PT_TOUCHPAD => {
+                            pointer_id = PointerId::MOUSE_ID;
+
+                            // There can be some rounding errors when doing the mouse hiMetric math,
+                            // and it doesn't make a whole lot of sense for a traditional mouse
+                            // pointer to be on a non-integer point, so round off any fractional
+                            // values that might've cropped up.
+                            position.x = position.x.round();
+                            position.y = position.y.round();
```

Comment:
> it doesn't make a whole lot of sense for a traditional mouse pointer to be on a non-integer point

I think this is a mistake. High-precision mouse pointers are widely supported and are useful for a number of purposes, e.g. in art applications. See prior discussion in https://github.com/rust-windowing/winit/pull/1375.

Reply:
High-precision mouse pointers are typically from pen/touch devices, which this rounding does not effect.

Reply:
Touchpads are a touch device. I'm not willing to assume without evidence that Windows always discards high-precision data from mice, either.

Reply:
Do you mean that touchpads are mouses actually? IIRC touchpads do have high precision data, but are treated as pointers on at least Wayland. So yeah, removing fractional part is likely not a good thing to do and I don't see much issue with it?

Reply:
Sorry, the terminology gets a bit confusing here. I mean that touchpads are high-precision input devices driven by a finger which can, should, and on non-Windows backends *do* have that precision propagated to the application. Even conventional mice can be, contrary to the comment above.

There's no reason a conventional mouse should be treated differently either, as you observe. Silently discarding perfectly good data is bad behavior.

Reply:
My mouse, standard (very high resolution) mouse also sends fractional parts when moving, and although it's not really useful in the windowing system with a standard pointer, it's very useful in games that support them.

Reply:
> although it's not really useful in the windowing system with a standard pointer

I'd like to emphasize that high-precision pointers absolutely can be useful. For example, any time you're interacting with a vector or zoomed-out raster graphics canvas, or operating a continuous UI control like a slider.

Reply:
> or operating a continuous UI control like a slider

Ah, true there, I have actually noticed that on some KDE sliders where I can barely adjust them but the cursor doesn't actually move...

Reply:
> Touchpads are a touch device. I'm not willing to assume without evidence that Windows always discards high-precision data from mice, either.

I have a laptop that uses the Windows Precision Touchpad drivers. When using the touchpad, Windows does not dispatch sub-pixel pointer movement events, and each movement event's position is within a `+-0.05` margin of a natural pixel number. The same is true with my mouse - I have a Logitech G502 gaming mouse, and Windows exhibits the same behavior there. This is a rounding error, and should be treated as such.

> My mouse, standard (very high resolution) mouse also sends fractional parts when moving, and although it's not really useful in the windowing system with a standard pointer, it's very useful in games that support them.

Are you on a Windows system? If so, could you check whether or not `event-polish` with this diff applied emits subpixel mouse movement events in the `window` example?

```diff
diff --git a/src/platform_impl/windows/event_loop.rs b/src/platform_impl/windows/event_loop.rs
index 3702d352..b1d7ae8c 100644
--- a/src/platform_impl/windows/event_loop.rs
+++ b/src/platform_impl/windows/event_loop.rs
@@ -1549,8 +1549,8 @@ unsafe extern "system" fn public_window_callback<T: 'static>(
                             // and it doesn't make a whole lot of sense for a traditional mouse
                             // pointer to be on a non-integer point, so round off any fractional
                             // values that might've cropped up.
-                            position.x = position.x.round();
-                            position.y = position.y.round();
+                            // position.x = position.x.round();
+                            // position.y = position.y.round();
 
                             buttons_down.set(
                                 PointerButton::MOUSE_LEFT.into_flags(),
```

Reply:
If it is ultimately determined that windows genuinely never reports non-integer mouse/touchpad coordinates, then at the very least the comment should reflect that, rather than claiming there's no use case for higher precision data. That way if a future change makes higher-precision data available there won't be cause for confusion as to whether it should be discarded.

Reply:
> Are you on a Windows system?

I'm not, linux with a custom mouse driver because my mouse is... unique.